SimTKcommon/src/Pathname.cpp

```diff
@@ -218,13 +241,13 @@ void Pathname::deconstructPathname( const string&   name,
     }
 
     // Now we can put together the canonicalized directory.
-    if (isAbsolutePath) {
-        if (!drive.empty())
-            directory = drive + ":";
+    if (dontApplySearchPath) {
```

Comment:
This seems confusing. `dontApplySearchPath` is an instruction to the caller, but we're not trying to search anything here. What is the real condition here? I think it might still be whether we have an absolute path. Then you can translate into a caller instruction just prior to returning, but this code is already complicated enough that I think the variable names should have the right meanings, even if you have to introduce an extra bool.